### ISSUE TITLE
[Feature request] Output attention scores in vLLM

### ISSUE THREAD

**Issue by ChenxinAn-fdu** [vllm-project/vllm#3192]
Thanks for the excellent work! 
I wonder can we get the attention scores when using vLLM?

**Comment by simon-mo** [comment]
This is currently not supported but a good feature request!

**Comment by rishsriv** [comment]
Hi there! Would you be open to a PR implementing attention scores? I would love to take this up if yes!

**Comment by havietisov** [comment]
Yeah it should. Would be useful for implementation of the Shrek sampler

**Comment by williammm001** [comment]
Any update on this?

**Comment by HuiSiqi** [comment]
Any update on this?

**Comment by HuiSiqi** [comment]
I also need to visualize the attentions scores.

**Comment by WoutDeRijck** [comment]
It would be insanse if this was made possible! @simon-mo, is this on the planning?

**Comment by OedoSoldier** [comment]
Is there any way to get attention scores? Like hook Q K or something?

**Comment by aciokler** [comment]
This feature would be very useful! Any updates?

**Comment by bruce2233** [comment]
This feature would be very useful! Any updates?

**Comment by OedoSoldier** [comment]
Guys, I've figured out that it's quite challenging to output the attention score, as vLLM employs a unique method to accelerate the attention calculation.

**Comment by nnurlan008** [comment]
Hi, 

I have enabled this feature using triton attention. Please take a look at this repo: https://github.com/nnurlan008/vllm.

**Comment by nnurlan008** [comment]
@mohamad-tohidi, my pleasure!
I think it is not that hard if you use the modified Triton backend. But you need to be careful about the attention matrix size, as it gets very large, and getting it through RPC from the client model executor might not be practical.

**Comment by hello-args** [comment]
A long-standing and highly requested feature.
